src/index.ts

```diff
@@ -13,49 +13,100 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
+import * as request from 'request';
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
 import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
```

Comment:
Any reason you're not using the [gcp-metadata](https://npm.im/gcp-metadata) module and querying the metadata service manually?

Reply:
I tried out gcp-metadata, and I found it took about 6 seconds to throw an error when I ran it locally.
There didn't seem to be a way to adjust the query/retries in gcp-metadata to make it faster, and 6 seconds felt like a long time to delay profiling when one is running outside of GCE.

Reply:
That's unexpected and should be investigated. If you have a minimal repro, can you open a bug against gcp-metadata?